app/controllers/students_controller.rb

```diff
@@ -0,0 +1,94 @@
+class StudentsController < ApplicationController
+    def index
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @students = @course.students
+    end
+
+    def new
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @student = @course.students.new
+    end
+
+    def create
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @student = @course.students.create(student_params)
+        if @student.save
+            redirect_to user_course_student_path(@user, @course, @student)
+        else
+            render 'new'
+      end
+    end
+
+    def show
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @student = @course.students.find(params[:id])
+    end
+
+    def destroy
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @student = @course.students.find(params[:id])
+        @student.destroy
+        redirect_to user_course_path(@user, @course)
+    end
+
+    def edit
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @student = @course.students.find(params[:id])
+    end
+
+    def update
+        @user = User.find(params[:user_id])
+        proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @student = @course.students.find(params[:id])
+
+        if @student.update(student_params)
+            redirect_to user_course_path(@user, @course)
+        else
+            render 'edit'
+        end
+    end
+
+    # Handles batch upload for students.
+    # The name of the file (minus the file extension) will be the name of the student.
+    def import
```

Comment:
Is this function called once per student, or once per upload? Can we make that a bit more explicit in either the function name or the comments?